drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/src/main/java/org/drools/workbench/screens/scenariosimulation/kogito/client/dmn/KogitoScenarioSimulationBuilder.java

```diff
@@ -0,0 +1,441 @@
+import org.drools.workbench.scenariosimulation.kogito.marshaller.js.SCESIMMainJs;
+import org.drools.workbench.scenariosimulation.kogito.marshaller.js.callbacks.SCESIMMarshallCallback;
+import org.drools.workbench.scenariosimulation.kogito.marshaller.js.model.JSIScenarioSimulationModelType;
+import org.drools.workbench.scenariosimulation.kogito.marshaller.js.model.SCESIM;
+import org.drools.workbench.scenariosimulation.kogito.marshaller.mapper.JSIName;
+import org.drools.workbench.scenariosimulation.kogito.marshaller.mapper.JsUtils;
+import org.drools.workbench.screens.scenariosimulation.kogito.client.converters.scesim.ApiJSInteropConverter;
+import org.drools.workbench.screens.scenariosimulation.model.typedescriptor.FactModelTree;
+import org.drools.workbench.screens.scenariosimulation.model.typedescriptor.FactModelTuple;
+import org.jboss.errai.common.client.api.RemoteCallback;
+import org.kie.workbench.common.dmn.webapp.kogito.marshaller.js.model.MainJs;
+import org.kie.workbench.common.dmn.webapp.kogito.marshaller.js.model.callbacks.DMN12UnmarshallCallback;
+import org.kie.workbench.common.dmn.webapp.kogito.marshaller.js.model.dmn12.JSITDefinitions;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.backend.vfs.PathFactory;
+
+import static org.drools.scenariosimulation.api.model.FactMappingType.EXPECT;
+import static org.drools.scenariosimulation.api.model.FactMappingType.GIVEN;
+import static org.drools.scenariosimulation.api.utils.ConstantsHolder.VALUE;
+
+/**
+ * Class used to populate <code>ScenarioSimulationModel</code> inside <code>Callback</code>s
+ * <p>
+ * Mostly copied from <code>org.drools.workbench.screens.scenariosimulation.backend.server.util.SimulationSettingsCreationStrategy</code> and its children.
+ * <p>
+ * Implemented to manage the nested callbacks for <b>DMN</b>
+ */
+public class KogitoScenarioSimulationBuilder {
+
+    @Inject
+    private KogitoDMNService dmnTypeService;
+
+    private static FactMappingType convert(final FactModelTree.Type modelTreeType) {
+        switch (modelTreeType) {
+            case INPUT:
+                return GIVEN;
+            case DECISION:
+                return EXPECT;
+            default:
+                throw new IllegalArgumentException("Impossible to map");
+        }
+    }
+
+    /**
+     * Populate the given <code>ScenarioSimulationModel</code> and returns it inn the given <code>callback</code>
+     * @param content
+     * @param type
+     * @param value
+     * @param callback
+     */
+    public void populateScenarioSimulationModel(final ScenarioSimulationModel content, final ScenarioSimulationModel.Type type, final String value, RemoteCallback<String> callback) {
+        switch (type) {
+            case RULE:
+                populateRULE(content, value, callback);
+                break;
+            case DMN:
+                populateDMN(content, value, callback);
+                break;
+            default:
+                // noop
```

Comment:
Maybe we should rather throw an exception?

Reply:
@jomarko  Done, thank you!